website/docs/opengl/c1/3-hello-texture.md

```diff
@@ -0,0 +1,388 @@
+* How to use texture parameters.
+* How to allow mipmaps.
+
+Like before, this tutorial will feature a lot of new content and info, so we'll take it slow and explain.
+
+## What is a texture
+Technically, textures are multidimensional objects that store color data.
+
+Mainly, textures store 2D data of the pixels of an image like a 2D array (this is just for comparison. Texture objects are a lot more
+complex than just 2D arrays!).
+This data can be used to be sanded to a shader, in an easy way for it can read.
+
+<?# Info "There's 1D or 3D texture objects, but we will focus on 2D" /?>
+
+But don't think texturing objects is as easy as a simple drag-and-drop! Textures need some information to be drawn.
+The most important information that we need to send to the texture to see it on the screen is the texture coordinates.
+The texture coordinates are the positions of the texture at a particular vertex. Below we will learn how to fill out
+this data and read it from a shader.
+
+
+## Sending more data to shaders
+With our new objective, let's learn how to send more data to shaders, and use it as texture coordinates.
+
+texture coordinates are a type of data that can be (and should be) manipulated by the vertex shader before be used by the fragment shader,
+so this is the way that we use to send the coordinates.
+
+Let's go back to our `OnLoad` function and update our `vertices` array to the following:
+```c#
+// The quad vertices data. Now with Texture coordinates!
+float[] vertices =
+{
+//       aPosition     | aTexCoords
+     0.5f,  0.5f, 0.0f,  1.0f, 1.0f,
+     0.5f, -0.5f, 0.0f,  1.0f, 0.0f,
+    -0.5f, -0.5f, 0.0f,  0.0f, 0.0f,
+    -0.5f,  0.5f, 0.0f,  0.0f, 1.0f
+};
+```
+
+With how we have structured the data, both the vertex and texture coordinates can be stored in a single array.
+
+If we try to run our program now...
+
+![A really messed up quad!](../../../images/opengl/chapter1/leasson3/messed-up-quad.png)
+
+"Oh no! We messed up with everything!" is what you should be thinking. And in reality, yes, we did. But it's simple to resolve, we just
+forgot to update our vertex layout definition.
+
+In our `VertexAttribPointer` calls, we declared a size of 3 and a stride of 3 for the `aPosition` attribute.
+Because of this, our buffer is being read like this:
+
+![Buffer reading wrong data](../../../images/opengl/chapter1/leasson3/wrong-pointers.png)
+
+To fix this, we just need to go back to our `VertexAttribPointer()`, which is below our shader compiling section, and
+change it to:
+```c#
+//                                        3 position + 2 texture coordinates! \/
+_gl.VertexAttribPointer(positionLoc, 3, VertexAttribPointerType.Float, false, 5 * sizeof(float), (void*)0);
+```
+
+This will make the buffer be read like that (including the texture coordinate pointer):
+![Buffer reading right data](../../../images/opengl/chapter1/leasson3/right-pointers.png)
+
+
+And if we run the program now, we can see our quad again!
```

Comment:
```suggestion
This will make the buffer be read like this (including the texture coordinate pointer, which we'll add into our code later):
![Buffer reading right data](../../../images/opengl/chapter1/leasson3/right-pointers.png)

Now the positions of each vertex are being read correctly! Each position is still three floats, but by skipping the two floats after each position, we skip over the U and V floats in between each XYZ floats, thus reading them correctly.

If we run the program now, we can see our quad again!
```

(By the way, when I write these suggestions adding additional paragraphs, these are just suggestions. If you'd like to word it differently you can discard what I wrote)

Reply:
anything you suggest will probably be better than anything that i can wrote so... lol